sunpy/net/dataretriever/sources/goes.py

```diff
@@ -168,9 +180,120 @@ def _can_handle_query(cls, *query):
+        satellitenumber = int(kwargs.get("satellitenumber", 16))
+        level = str(kwargs.get("level", "2"))  # cast to string because it may be provided as number
+
+        if level == "2":
+            search_pattern = base_url + 'l{level}/data/suvi-l{level}-ci{wave:03}/%Y/%m/%d/dr_suvi-l{level}-ci{wave:03}_g16_s%Y%m%dT%H%M%SZ_.*\.fits'
+        elif level == "1b":
+            if wave in [131, 171, 195, 284]:
+                search_pattern = base_url + 'l{level}/suvi-l{level}-fe{wave:03}/%Y/%m/%d/OR_SUVI-L{level}-Fe{wave:03}_G16_s%Y%j%H%M%S.*\.fits.gz'
+            elif wave == 304:
+                search_pattern = base_url + 'l{level}/suvi-l{level}-he{wave:03}/%Y/%m/%d/OR_SUVI-L{level}-He{wave_minus1:03}_G16_s%Y%j%H%M%S.*\.fits.gz'
+            elif wave == 94:
+                search_pattern = base_url + 'l{level}/suvi-l{level}-fe{wave:03}/%Y/%m/%d/OR_SUVI-L{level}-Fe{wave_minus1:03}_G16_s%Y%j%H%M%S.*\.fits.gz'
+        else:  # should never reach here if called by dataretriever
+            return []
+
+        if search_pattern.count('wave_minus1'):
+            scraper = Scraper(search_pattern, level=level, wave=wave,
+                              goes_number=satellitenumber, wave_minus1=wave-1)
+        else:
+            scraper = Scraper(search_pattern, level=level, wave=wave,
+                              goes_number=satellitenumber)
+        return scraper.filelist(timerange)
+
+    def _makeimap(self):
+        """
+        Helper Function used to hold information about source.
+        """
+        self.map_['source'] = 'GOES'
+        self.map_['provider'] = 'NOAA'
+        self.map_['instrument'] = 'SUVI'
+        self.map_['physobs'] = 'flux'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+        """
+        Answers whether client can service the query.
+
+        Parameters
+        ----------
+        query : `tuple`
+            All specified query objects
+
+        Returns
+        -------
+        `bool`
+            answer as to whether client can service the query
+        """
+        supported_levels = ("2", "1b")
+        supported_waves = [94, 131, 171, 195, 284, 304]
+        minimum_supported_satellitenumber = 16
+
+        required = {a.Time, a.Instrument, a.Wavelength}
+        optional = {} #, a.SatelliteNumber, a.Level}
+        all_attrs = {type(x) for x in query}
+
+        ops = all_attrs - required
+        # If ops is empty or equal to optional we are ok, otherwise we don't
+        # match
+        if ops and ops != optional:
+            return False
+
+        # if we get this far we have either Instrument and Time
+        # or Instrument, Time and Wavelength
+        check_var_count = 0
+        for x in query:
+            if isinstance(x, a.Instrument) and x.value.lower() == 'suvi':
+                check_var_count += 1
+            if isinstance(a, a.Wavelength) and int(x.value.to_value('Angstrom')) is in supported_waves:
```

Comment:
so this is no longer checked in this function as we discussed and agreed. How does this range get passed to `_get_url_for_timerange`?

Reply:
I honestly can't remember, try it and find out!

*I would guess it's a tuple*

Reply:
It's a minmax object but only if it is called `a.Wavelength(q1, q2)` otherwise the bare quantity object get passed. This is weird. and probably very few clients are currently handling this. Would make more sense to always pass a `minmax` object but with the same values in both.

Reply:
Why does it pass the Wavelength attr object directly?!

Reply:
`minmax` is a named tuple. 
```
namedtuple("minmax", "{0}min {0}max".format(prefix))
```